### PR TITLE
Introduce user queries objects

### PR DESCRIPTION
**Pull request by aledeg**
There is now an object to manipulate user queries. It allows to move logic to handle those from the view and the controller in the model.
Thus making the view and the controller easier to read.
I introduced a new interface to start using dependency injection. There is still some rough edges but we are moving in the right direction.
The new object is fully tested but it still need some improvements, for instance, it is still tied to the search object.
There might be a better way to do that.

### CODE REVIEW

**Comment by aledeg** [comment]
I made a new PR and I squash all my small commits into one.
It make things clearer in the git history.
I included everything we talk about in the other PR. I should be ok now.

File: app/Exceptions/DAOException.php [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<?php
+
+class FreshRSS_DAOException extends \Exception {
```
Comment (marienfressinaud) [inline app/Exceptions/DAOException.php]: Could you add an underscore after `DAO`?

File: app/Models/CategoryDAO.php [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <?php
 
-class FreshRSS_CategoryDAO extends Minz_ModelPdo {
+class FreshRSS_CategoryDAO extends Minz_ModelPdo implements FreshRSS_Searchable{
```
Comment (marienfressinaud) [inline app/Models/CategoryDAO.php]: There is a missing space before `{`

File: app/Models/EntryDAO.php [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <?php
 
-class FreshRSS_EntryDAO extends Minz_ModelPdo {
+class FreshRSS_EntryDAO extends Minz_ModelPdo implements FreshRSS_Searchable{
```
Comment (marienfressinaud) [inline app/Models/EntryDAO.php]: Same here

File: app/Models/FeedDAO.php [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <?php
 
-class FreshRSS_FeedDAO extends Minz_ModelPdo {
+class FreshRSS_FeedDAO extends Minz_ModelPdo implements FreshRSS_Searchable{
```
Comment (marienfressinaud) [inline app/Models/FeedDAO.php]: Same here

File: app/Models/UserQuery.php [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+
+	/**
+	 * Parse the query string when it is a "category" query
+	 *
+	 * @param integer $id
+	 * @throws FreshRSS_DAOException
+	 */
+	private function parseCategory($id) {
+		if (is_null($this->category_dao)) {
+			throw new FreshRSS_DAOException('Category DAO is not loaded i UserQuery');
```
Comment (marienfressinaud) [inline app/Models/UserQuery.php]: "i" UserQuery?

File: app/Models/UserQuery.php [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+
+	/**
+	 * Parse the query string when it is a "feed" query
+	 *
+	 * @param integer $id
+	 * @throws FreshRSS_DAOException
+	 */
+	private function parseFeed($id) {
+		if (is_null($this->feed_dao)) {
+			throw new FreshRSS_DAOException('Feed DAO is not loaded i UserQuery');
```
Comment (marienfressinaud) [inline app/Models/UserQuery.php]: Same here

**Comment by marienfressinaud** [comment]
Sorry I missed a bunch of details yesterday.

I'm not fan of squashing commits: we lose the work process (i.e. how the work has been done). The main argument is usually to clean history but it seems to me that is the role of creating a dedicated branch.

**Comment by aledeg** [comment]
I agree for the work process but when it comes for minor changes, I find that it is too verbose.
My two cents.

**Comment by marienfressinaud** [comment]
I mess up a bit my merging but everything seems to work.

**Comment by marienfressinaud** [comment]
I merged your work locally and rebased it on your latest changes (#803) and push it so it is the reason the ticket has not been merged properly.

**Pull request closed without merging**